Improve VTOrc config handling to support dynamic variables (#17218)
vitessio/vitess: changelog/22.0/22.0.0/summary.md

These are the RPC changes made in this release - 

1. `GetTransactionInfo` RPC has been added to both `VtctldServer`, and `TabletManagerClient` interface. These RPCs are used to fecilitate the users in reading the state of an unresolved distributed transaction. This can be useful in debugging what went wrong and how to fix the problem.

### <a id="vtorc-config-file-changes"/>VTOrc Config File Changes</a>

The configuration file for VTOrc has been updated to now support dynamic fields. The old `--config` parameter has been removed. The alternative is to use the `--config-file` parameter. The configuration can now be provided in both json, yaml or any other format that [viper](https://github.com/spf13/viper) supports.

The following fields can be dynamically changed - 
1. `InstancePollTime`
2. `PreventCrossCellFailover`
3. `SnapshotTopologyInterval`
4. `ReasonableReplicationLag`
5. `AuditToBackend`
6. `AuditToSyslog`
7. `AuditPurgeDuration`
8. `WaitReplicasTimeout`
9. `TolerableReplicationLag`
10. `TopoInformationRefreshDuration`

> 11. `RecoveryPollDuration`

Reviewer: We should add more flags to dynamic handling, there's no reason not to. One that comes to mind is `--allow-emergency-reparent`.

Author: Done

> To upgrade to the newer version of the configuration file, the users can first change to using the flags in the previous release before upgrading. They can then revert to using the configuration file in the newer release.

Reviewer: ```suggestion
To upgrade to the newer version of the configuration file, first switch to using the flags in your current deployment before upgrading. Then you can switch to using the configuration file in the newer release.
```

Author: Fixed

valid candidates. This means they will never get promoted - not even if there's no other candidates.

Note that behavior for `builtin` backups remains unchanged: a replica that is currently taking a `builtin` backup will
never be promoted, neither by planned nor by emergency reparents.

### <a id="vtorc-config-file-changes"/>VTOrc Config File Changes</a>

> The configuration file for VTOrc has been updated to now support dynamic fields. The old `--config` parameter has been removed. The alternative is to use the `--config-file` parameter. The configuration can now be provided in both json, yaml or any other format that [viper](https://github.com/spf13/viper) supports.

Reviewer: ```suggestion
The configuration file for VTOrc has been updated to now support dynamic fields. The old `--config` parameter has been removed. The alternative is to use the `--config-file` parameter. The configuration can now be provided in json, yaml or any other format that [viper](https://github.com/spf13/viper) supports.
```
